src/storage/gc_worker.rs

```diff
@@ -0,0 +1,172 @@
+// Copyright 2018 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use super::engine::{Engine, ScanMode};
+use super::mvcc::{MvccReader, MvccTxn, MAX_TXN_WRITE_SIZE};
+use super::txn::GC_BATCH_SIZE;
+use super::{Callback, Error, Key, Result};
+use kvproto::kvrpcpb::Context;
+use std::fmt::{self, Display, Formatter};
+use std::sync::{Arc, Mutex};
+use util::worker::{self, Builder, Runnable, Worker};
+
+struct GCTask {
+    pub ctx: Context,
+    pub safe_point: u64,
+    pub callback: Callback<()>,
+}
+
+impl Display for GCTask {
+    fn fmt(&self, f: &mut Formatter) -> fmt::Result {
+        let epoch = match self.ctx.region_epoch.as_ref() {
```

Comment:
can we meet a case that the region has no epoch?

Reply:
It seems not, but the fmt method shouldn't panic when it meets logically incorrect value.